cmd/scollector/collectors/azureeabilling.go

```
@@ -0,0 +1,192 @@
+			ioutil.WriteFile("c:\\temp\\ea-price.csv", []byte(csv.PriceSheetReport), 0755)
+			ioutil.WriteFile("c:\\temp\\ea-detail.csv", []byte(csv.DetailReport), 0755)
+			structs, err := csv.ConvertToStructs()
+
+			if err != nil {
+				return err
+			}
+			for _, p := range structs.PriceSheetReport {
+				err := processAzureEAPriceSheetRow(p, md)
+				if err != nil {
+					return err
+				}
+			}
+			for _, d := range structs.DetailReport {
+				err := processAzureEADetailRow(d, md)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// processAzureEAPriceSheetRow will take the price sheet info and log it, so we can track price changes over time
+func processAzureEAPriceSheetRow(p *azureeabilling.PriceSheetRow, md *opentsdb.MultiDataPoint) error {
+	fullProdName := fmt.Sprintf("%s-%s", p.Service, p.UnitOfMeasure)
+	priceString := convertAzurePriceToString(p.UnitPrice)
+	tags := opentsdb.TagSet{
+		"partnumber": p.PartNumber,
+		"service":    fullProdName,
+	}
+	Add(md, "azure.ea.pricesheet", priceString, tags, metadata.Gauge, metadata.Count, priceDesc)
+	return nil
+}
+
+// processAzureEADetailRow will take the actual usage data for the provided month
+func processAzureEADetailRow(p *azureeabilling.DetailRow, md *opentsdb.MultiDataPoint) error {
+	//Don't process todays records as they are subject to change
+	nowYear, nowMonth, nowDay := time.Now().Date()
+	recordMonth := int(nowMonth)
+	if nowYear == p.Year && recordMonth == p.Month && nowDay == p.Day {
+		return nil
+	}
+
+	resourcePaths := strings.Split(strings.ToLower(p.InstanceID), "/")
+	var resourceString string
+
+	if len(resourcePaths) < 8 {
+		resourceString = strings.ToLower(p.InstanceID)
+	} else {
+		resourceIDs := resourcePaths[8:]
+		resourceString = strings.Join(resourceIDs, "-")
+	}
+
+	if p.ResourceGroup != "" {
+		resourceString = fmt.Sprintf("%s-%s", strings.ToLower(p.ResourceGroup), resourceString)
+	}
+
+	resourceString, _ = opentsdb.Clean(resourceString)
```

ignoring err intentional, if so comment maybe